shuup/core/models/_contacts.py

```diff
@@ -454,12 +459,31 @@ def get_company_contact(user):
       CompanyContact (or none) of which user's PersonContact is a member
     :rtype: CompanyContact|None
     """
+    if should_force_person_contact(user):
+        return None
+
     contact = get_person_contact(user)
     if not contact:
         return None
     return contact.company_memberships.filter(is_active=True).first()
 
 
+def get_company_contact_for_shop(shop):
+    return CompanyContact.objects.get_or_create(
```

Comment:
Yeah likely useful to add the shop. But I guess it wouldn't hurt if this would have been more behind the scenes user that is not listed. For now I think I add this.